### DISCUSSION TITLE
Searching for usages without leaving all scenes open afterwards

### DISCUSSION THREAD

**Discussion by mikkellam** [yasirkula/UnityAssetUsageDetector#38] [Q&A] [answered]
Hey! I've been using your asset for quite a while and it really makes it a lot easier to clean up asset usages or ensuring you are not removing an asset that was used in a place you did not realize. I have however recently move on to a project with significantly higher fidelity assets and quite big scenes. So whenever I do a search for an asset I thought not too widely used and it turns out it was used in a lot of scenes, I am left with a large quantity of massive scenes open and stuck with the editor freezing up for 3-4 minutes every editor update such that I have to hit some 1-2 second timing between updates to reset the scenes and actually be able to use unity again. This kind of makes it too risky to effectively use the asset like i used to on the smaller project since every time is a risk of me being stuck for 10-20 minutes trying to get out of the freeze loop.

So I was wondering if there was an option to search for usages without leaving all the scenes the usages are in open? I can't seem to find it in the settings as is.

**Comment by yasirkula** [comment, marked as answer]
Hi! If you enable Lazy scene search, it won't open any closed scenes automatically. It'll just tell you that the scene has potential references in it and search the scene in detail only if you click the Refresh button next to it. This feature is unfortunately disabled if "Addressables support" is enabled in the settings because Unity doesn't tell us in advance if an asset is used in a scene (Lazy scene search relies on it) if the asset was referenced via an AssetReference.

**Reply by mikkellam** [reply]
I somehow missed lazy scene search as being a simillar solution. That works perfectly then i can see when its potentially gonna open 20 scenes and even go through them one at a time if I was of a mind to clean them all up. 

Thanks a lot! ^^
